### ISSUE TITLE
performing self-learning and accuracy measuring on Imagenet-A

### ISSUE THREAD

**Issue by Djoels** [bethgelab/robustness#8]
Does it make sense to perform self-learning on Imagenet-A? I tried implementing it but somehow the accuracy never goes above 0.2%

my code looks like this after combining Imagenet-C code from here
https://github.com/bethgelab/robustness/blob/main/examples/selflearning/main.py
 with the eval-code from 
https://github.com/hendrycks/natural-adv-examples/blob/master/eval.py

now looking like this:

```
    imga = ImageNetA() # as defined here https://github.com/bethgelab/robustness/blob/main/robusta/datasets/imageneta.py
    # but fixed every attribute to self.attribute
    for epoch in range(num_epochs):
        predictions = []
        for images, targets in val_loader:

            logits = model(images.cuda())[:,imga.indices_in_1k] # fixed for correct accuracy metric?
            predictions = logits.data.max(1)[1]
            loss = gce(logits, predictions, q = gce_q)

            optimizer.zero_grad()
            loss.backward()
            optimizer.step()

            num_correct += predictions.eq(targets.data).sum().item()
            confidence.extend(to_np(F.softmax(logits, dim=1).max(1)[0]).squeeze().tolist())
            correct.extend(predictions.eq(targets).to('cpu').numpy().squeeze().tolist())
            num_samples += len(targets)
            print(f"Correct: {num_correct:#5.0f}/{num_samples:#5.0f} ({100 * num_correct / num_samples:.2f} %)")
```

Is my approach wrong? should I use an adapted accuracy metric like in imagenet-R?
I believe I had fixed for that with the line of code only retaining the relevant logits (see above)...

**Comment by EvgeniaAR** [comment]
Sorry for the late reply! It depends which model you are using for it. In case of a ResNet50, I would say it does not make sense. Self-learning works by reinforcing the predictions, so they need to be at least somewhat correct. ImageNet-A is designed to fool a ResNet50, so making it more confident in its wrong predictions will not make the accuracy better. Your code looks correct to me.

**Comment by Djoels** [comment]
Hi, apologies for being late to  closing this. I guess my understanding  was wrong, thank you for giving an intuition into why :)
